packages/node-runtime-worker-thread/src/index.ts

```diff
@@ -11,184 +9,162 @@ import type {
 import type { MongoshBus } from '@mongosh/types';
 import path from 'path';
 import { EventEmitter, once } from 'events';
-import { kill } from './spawn-child-from-source';
 import type { Caller } from './rpc';
-import { createCaller, cancel } from './rpc';
-import { ChildProcessEvaluationListener } from './child-process-evaluation-listener';
+import { createCaller, cancel, exposeAll } from './rpc';
 import type { WorkerRuntime as WorkerThreadWorkerRuntime } from './worker-runtime';
 import {
   deserializeEvaluationResult,
   serializeConnectOptions,
 } from './serializer';
-import { ChildProcessMongoshBus } from './child-process-mongosh-bus';
 import type { CompassServiceProvider } from '@mongosh/service-provider-server';
+import { SHARE_ENV, Worker } from 'worker_threads';
```

Comment:
@Anemy iirc what we tried to use in Compass were Web Workers with `nodeIntegrationInWorker: true` ... does this work?

Reply:
Hmmm adding that option here: https://github.com/mongodb-js/compass/blob/66e4d196949cb48787fde3c582b2e3d052ac103a/packages/compass/src/main/window-manager.ts#L124 
On a quick test pulling this code into Compass is getting me:
<img width="606" alt="Screenshot 2024-07-26 at 10 15 51 AM" src="https://github.com/user-attachments/assets/8eb2d9cb-aeae-404c-9372-f09d9d2f1c31">

```
The V8 platform used by this instance of Node does not support creating Workers
Call Stack
 new Worker
  node:internal/worker:213:21
 WorkerRuntime.initWorker
  MongoDBCompass/../compass-shell/src/node-runtime-worker-thread/index.ts:62:27
```

Branch: test-poc-worker
https://github.com/mongodb-js/compass/compare/main...test-poc-worker

Reply:
Yeah – Electron doesn't support Node.js `worker_threads` (hence the idea to use Web Workers instead)

Reply:
Yep, that was the reason we had a child_process added in the initial implementation: to work around Electron runtimes not supporting those

Reply:
TIL, thanks, I'll update this to web workers and see if things work smoothly.

Reply:
Updated https://github.com/mongodb-js/compass/compare/main...test-poc-worker to use a web worker, appears to run smoothly, woo! Updating this pr next and will see how we can test it.